### ISSUE TITLE
Build fail on iOS v1.41

### ISSUE THREAD

**Issue by erf** [RevenueCat/purchases-flutter#123]
## Common project specific issues

**Describe the bug**

I fail to build on iOS on latest version v1.4.1

1. Environment
    1. Platform: iOS
    2. SDK version: Stable 1.22.2
    3. OS version: 14.0.1
    4. XCode/Android Studio version: 12.1 / 4.1

**Additional context**

I tried to remove the `Podfile.lock` and `Pods` folder and did a `flutter clean` and `flutter install` before calling `pod install` manuallly in the `ios` folder. Still no go. 

It was working on version `1.4.0`, but i had some build issues there also and had to regularly remove the `Podfile.lock` and the ` Pods` folder and reinstall manually.

```
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    ld: warning: dylib
    (/Users/erlend/projects/my_project/build/ios/Profile-iphoneos/PromisesObjC/FBLPromises.framework/F
    BLPromises) was built for newer iOS version (9.0) than being linked (8.0)
    /Users/erlend/projects/my_project/ios/Pods/PurchasesHybridCommon/ios/PurchasesHybridCommon/Purchas
    esHybridCommon/RCPurchases+HybridAdditions.m:13:17: warning: method definition for
    '_setPushTokenString:' not found [-Wincomplete-implementation]
    @implementation RCPurchases (HybridAdditions)
                    ^
    In file included from
    /Users/erlend/projects/my_project/ios/Pods/PurchasesHybridCommon/ios/PurchasesHybridCommon/Purchas
    esHybridCommon/RCPurchases+HybridAdditions.m:6:
    /Users/erlend/projects/my_project/ios/Pods/PurchasesHybridCommon/ios/PurchasesHybridCommon/Purchas
    esHybridCommon/RCPurchases+HybridAdditions.h:14:1: note: method '_setPushTokenString:' declared
    here
    - (void)_setPushTokenString:(nullable NSString *)pushToken;
    ^
    /Users/erlend/projects/my_project/ios/Pods/PurchasesHybridCommon/ios/PurchasesHybridCommon/Purchas
    esHybridCommon/RCPurchases+HybridAdditions.m:13:17: warning: method definition for
    'configureWithAPIKey:appUserID:observerMode:userDefaults:platformFlavor:platformFlavorVersion:'
    not found [-Wincomplete-implementation]
    @implementation RCPurchases (HybridAdditions)
                    ^
    In file included from
    /Users/erlend/projects/my_project/ios/Pods/PurchasesHybridCommon/ios/PurchasesHybridCommon/Purchas
    esHybridCommon/RCPurchases+HybridAdditions.m:6:
    /Users/erlend/projects/my_project/ios/Pods/PurchasesHybridCommon/ios/PurchasesHybridCommon/Purchas
    esHybridCommon/RCPurchases+HybridAdditions.h:23:1: note: method
    'configureWithAPIKey:appUserID:observerMode:userDefaults:platformFlavor:platformFlavorVersion:'
    declared here
    + (instancetype)configureWithAPIKey:(NSString *)APIKey
    ^
    2 warnings generated.
    /Users/erlend/.pub-cache/hosted/pub.dartlang.org/purchases_flutter-1.4.1/ios/Classes/PurchasesFl
    utterPlugin.m:197:28: warning: 'addAttributionData:network:networkUserId:' is deprecated: Use
    the set<NetworkId> functions instead. [-Wdeprecated-declarations]
        [RCCommonFunctionality addAttributionData:data network:network networkUserId:networkUserId];
                               ^
    In module 'PurchasesHybridCommon' imported from
    /Users/erlend/.pub-cache/hosted/pub.dartlang.org/purchases_flutter-1.4.1/ios/Classes/PurchasesFl
    utterPlugin.m:5:
    /Users/erlend/projects/my_project/ios/Pods/PurchasesHybridCommon/ios/PurchasesHybridCommon/Purchas
    esHybridCommon/RCCommonFunctionality.h:22:130: note: 'addAttributionData:network:networkUserId:'
    has been explicitly marked deprecated here
    + (void)addAttributionData:(NSDictionary *)data network:(NSInteger)network
    networkUserId:(NSString *) networkUserId __attribute((deprecated("Use the set<NetworkId>
    functions instead.")));
                                                                                                                                     ^
    1 warning generated.
    ld: warning: dylib
    (/Users/erlend/projects/my_project/build/ios/Profile-iphoneos/GTMSessionFetcher/GTMSessionFetcher.
    framework/GTMSessionFetcher) was built for newer iOS version (9.0) than being linked (8.0)
    Undefined symbols for architecture arm64:
      "_OBJC_CLASS_$_RCPurchases", referenced from:
          objc-class-ref in PurchasesFlutterPlugin.o
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
```

**Comment by aboedo** [comment]
the only difference between `1.4.0` and `1.4.1` is that linking is no longer done statically (https://github.com/RevenueCat/purchases-ios/pull/374), but you should still be able to get the old behavior by adding 
`use_frameworks! :linkage => :static` to your podfile (or editing the `use_frameworks!` line if it's already there). Could you give that a try?

**Comment by erf** [comment]
It worked to add your suggestion like this:

```
target 'Runner' do
  #use_frameworks!
  use_frameworks! :linkage => :static
```

I also did remove Pods and Podfile.lock and reinstalled them manually.

I hope you can fix it so i don't have to edit my Podfile to run a Flutter app. Thanks!

**Comment by aboedo** [comment]
@erf I'm glad it's working now. see my comment here, though: https://github.com/RevenueCat/purchases-flutter/issues/120#issuecomment-716663133. I think this may be an Xcode 12 bug.

**Comment by erf** [comment]
@aboedo @rkotzy why was the issue closed? Do you consider the problem solved as it worked by editing the Podfile? Or do you blame XCode 12 for this? The other plugins are working fine, so it seem like the problem is with this plugin..?

**Comment by aboedo** [comment]
@erf I'm not sure yet why this is happening. 
It does help to know that the static linking fixed it for you, but it might also be an indicator that there are some cached objects lying around somewhere which the linker was trying and failing to use. Could you try building in xcode 11 and then xcode 12 again? 

This ticket is a duplicate of #120, so let's keep the conversation there.

**Comment by erf** [comment]
Ah, so a duplicate - good to let me know.

I'm not able to test in XCode 11 now, but i saw you was able to build in 12 now. Let's move the conversation there.
